### PR TITLE
chore: apply `apply` instead of `refine`

### PR DESCRIPTION
**Pull request by mo271**
We replace `refine foo ?_ ?_` by `apply foo`, and similar cases in order to avoid many underscores.
For instance
```
refine' liftR_map_last _ _ _ _ _
```
is replaced with 
```
apply liftR_map_last
```
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by mo271** [comment]
!bench

File: Mathlib/Algebra/Module/Zlattice/Basic.lean [hunk 1]
```diff
@@ -522,7 +522,7 @@ theorem Zlattice.rank [hs : IsZlattice K L] : finrank ℤ L = finrank K E := by
     exact Subtype.coe_injective.comp (Basis.injective _)
   rw [finrank_eq_card_chooseBasisIndex]
     -- We prove that `finrank ℤ L ≤ finrank K E` and `finrank K E ≤ finrank ℤ L`
-  refine le_antisymm ?_ ?_
+  apply le_antisymm
```
Comment (dupuisf) [inline Mathlib/Algebra/Module/Zlattice/Basic.lean:525]: I don't think this should be changed. A `refine` call with two or more `?_` holes tells the readers how many new goals will be created. If I see `apply`, I would only expect one goal to be left afterwards.

Comment (mo271) [reply Mathlib/Algebra/Module/Zlattice/Basic.lean:525]: There are 2745 matches for `apply.*\n *·`, so it seems in those cases there is more than one goal left.

Comment (mo271) [reply Mathlib/Algebra/Module/Zlattice/Basic.lean:525]: The reader can also see how many new goals are created by looking at the well structured indented "`·`".

**Comment by mo271** [comment]
This seems more controversial than I anticipated, see
https://leanprover.zulipchat.com/#narrow/stream/144837-PR-reviews/topic/.2312259.20more.20.60apply.60

**Comment by mo271** [comment]
closing in favour of doing only the one_underscore part of this here:
https://github.com/leanprover-community/mathlib4/pull/12331

**Pull request closed without merging**